decoder/decoder_test.go

```diff
@@ -94,3 +96,38 @@ func TestDecodeSizeLimitGzip(t *testing.T) {
 	assert.NotNil(t, err)
 	assert.NotEqual(t, err, io.EOF)
 }
+
+func TestDecodeSourcemapFormData(t *testing.T) {
+
+	body := &bytes.Buffer{}
+	writer := multipart.NewWriter(body)
+
+	fileBytes, err := loader.LoadDataAsBytes("data/valid/sourcemap/bundle.js.map")
+	assert.NoError(t, err)
+	part, err := writer.CreateFormFile("sourcemap", "bundle_no_mapping.js.map")
+	assert.NoError(t, err)
+	_, err = io.Copy(part, bytes.NewReader(fileBytes))
+	assert.NoError(t, err)
+
+	writer.WriteField("bundle_filepath", "js/./test/../bundle_no_mapping.js.map")
+	writer.WriteField("service_name", "My service")
+	writer.WriteField("service_version", "0.1")
+
+	err = writer.Close()
+	assert.NoError(t, err)
+
+	req, err := http.NewRequest("POST", "_", body)
+	req.Header.Set("Content-Type", writer.FormDataContentType())
+	assert.NoError(t, err)
+
+	assert.NoError(t, err)
+	data, err := DecodeSourcemapFormData(req)
+	assert.NoError(t, err)
+
+	assert.Len(t, data, 4)
+	assert.Equal(t, "js/bundle_no_mapping.js.map", data["bundle_filepath"])
+	assert.Equal(t, "My service", data["service_name"])
+	assert.Equal(t, "0.1", data["service_version"])
+	assert.NotNil(t, data["sourcemap"].(string))
+	assert.Equal(t, len(fileBytes), len(data["sourcemap"].(string)))
+}
```

Comment:
I really like this change to havedecoding functions in one place! Could you please also add some basic tests for the other decode data?

Reply:
Done! c0649a7e0f2ed7d7407ce3cd3d683de86920da71